### ISSUE TITLE
Split into two branches: One that supports PowerShell V2 and another that does not

### ISSUE THREAD

**Issue by proxb** [proxb/PoshRSJob#162]
I'm looking at creating two branches so we can focus development of this module that only supports V3+ and another that allows for V2+ support. Supporting V2 is sometimes challenging and requires extra code to make things that are normally simple in V3+ to implement. We would still support both branches with the difference that the V2+ branch would not be available on the PowerShell Gallery.

**Comment by MVKozlov** [comment]
after my PR I hope ;-)

**Comment by proxb** [comment]
Of course. I'm going to spend a little time this weekend doing some "catch up" to approve PRs and then begin setting up the branches.

**Comment by MVKozlov** [comment]
My fork have two possible PR's, One - which I'm already opened and another, in separate branch.

It have parameter/argumentlist optimizations and several different ways of handling (#149).
For compatibility I can switch it to 'same' before submitting it.
But seems there can be discussions about, or at least it should be well documented (#160)

**Comment by proxb** [comment]
I think this last version (1.7.4.2) will be the last version to support V2. I'll split this off onto a V2 branch that will exist just for those who need V2 support and this main branch will be supporting V3+ which means I get to remove all of that fun V2 code :)

**Comment by proxb** [comment]
I created a [new branch](https://github.com/proxb/PoshRSJob/tree/PowerShellV2) for the V2 side. Next up is to push a new Master branch that only supports V3+.

**Comment by AspenForester** [comment]
I was going to post a really minor issue, but it seems appropriate for this thread.  1.7.4.4 is still using the `ModuleToProcess` manifest member which generates a warning when I publish to my local NuGet gallery.  The preferred manifest member is `RootModule`.  It doesn't feel like that really warrants an issue to itself.  And for one line in the psd1, a fork and pull request even seems a little over kill.

**Comment by proxb** [comment]
@AspenForester I can make that update and roll it in with the 1.7.5.0 update. Thanks for pointing that out!
